wiki/Ranking_Criteria/osu!taiko/en.md

```diff
@@ -3,46 +3,161 @@ osu!taiko
+
+#### Rules
+- **Patterns on snaps faster than 1/2 are disallowed.** These patterns are too complex for beginner players.
+
+#### Guidelines
+- **1/2 patterns should not be longer than four notes.**
+- **Slider velocity changes may be used cautiously.** Changes should only happen for sections of different pacings and slider velocity should not be drastically variable. 
+
+#### Difficulty Setting Guidelines
+- OD should be 5 or less.
+- HP should be 5 or more. In cases of higher note count, HP could be slightly adjusted to lower than 5.
+
+### ![Muzukashii](/wiki/shared/Hard-t.png "Muzukashii") Muzukashii
+
+#### Rules
+- **Patterns on snaps faster than 1/6 are disallowed.** These patterns are too complex for the target audience of this difficulty level.
+- **Finisher notes must not be used in any 1/4 patterns in this difficulty.** These patterns are not suitable for this difficulty level. 
+
+#### Guidelines
+- **1/6 patterns should not be longer than four notes on moderate to low BPM (~140).** Anything longer is very situational and normally too complex. Patterns like this should be followed by a rest moment and avoided at faster BPMs. 
+- **1/4 patterns should not be longer than five notes.** Anything longer is likely to be too straining for intermediate players. 
+- **Main snapping should consist of mostly 1/2, 1/1 or slower rhythms.** Occasionally using ¼ rhythms is acceptable.
+- **Try to insert at least 1 rest moment that is 3/2 or longer after 16/1 to 20/1 of continuous mapping.** Less frequent rest moments or shorter ones may put too much strain on intermediate players.
+- **Manipulating slider velocity is allowed**, but slider velocity should only be changed for sections of music with different pacings and should not be drastically variable.
+- **1/4 patterns with one or more color changes should be used sparingly.** They should be avoided in conjunction with other patterns of this nature because the target audience of this difficulty level is not used to patterns of this complexity. 
+- **1/4 patterns which are longer than three notes should consist of one color change at most which should take place at the start or the end of the pattern.** Patterns more complex than that would be too demanding for intermediate players. These patterns should be followed by a rest moment. 
+
+#### Difficulty Setting Guidelines
+- OD should be 5 or less
+- HP should be 5 or more. In cases of higher note count, HP could be slightly adjusted to lower than 5.
+
+### ![Oni](/wiki/shared/Insane-t.png "Oni") Oni
+
+#### Rules
+- **Patterns on snaps faster than 1/8 are disallowed.** These patterns are too complex for the target audience of this difficulty level.
+
+#### Guidelines
+- **1/8 patterns should not be longer than two notes.** It may be used for songs which follow a swing beat. Anything longer is very situational and normally too complex. Patterns like this should be followed by a rest moment.
```

Comment:
1/8 would be incorrect snapping for a swing beat song, so I'm confused as to what this actually wants to say